fetch.js

```diff
@@ -236,6 +236,21 @@
       }
     }
 
+    if (support.arrayBuffer) {
+      this.arrayBuffer = function() {
+        if (this._bodyArrayBuffer) {
+          var rejected = consumed(this)
+          if (rejected) {
+            return rejected
```

Comment:
I'm fine with this refactor, but this pattern is everywhere in this file. Doesn't seem appropriate to hold this change up over that style debate.